enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1392,14 +1393,20 @@ func (ws *workspace) sync(ctx context.Context) error {
 	if *pushedRepoURL != "" {
 		checkoutRef = fmt.Sprintf("%s/%s", gitRemoteName(*pushedRepoURL), *pushedBranch)
 		checkoutLocalBranchName = *pushedBranch
+	} else if *targetBranch != "" {
+		checkoutRef = fmt.Sprintf("%s/%s", gitRemoteName(*targetRepoURL), *targetBranch)
+		checkoutLocalBranchName = *targetBranch
 	} else {
-		if *targetBranch != "" {
-			checkoutRef = fmt.Sprintf("%s/%s", gitRemoteName(*targetRepoURL), *targetBranch)
-			checkoutLocalBranchName = *targetBranch
-		} else {
-			checkoutRef = *targetCommitSHA
-			checkoutLocalBranchName = "local"
-		}
+		// If no branch is passed in, stay on the default branch. Don't set checkoutRef
+		checkoutLocalBranchName = "local"
+	}
+
+	// TODO(Maggie): If commit sha is not set, pull the actual sha so that it can be used in reporting
+	checkoutCommit := "HEAD"
```

Comment:
I think `HEAD` is not the right default value - HEAD points to the currently checked out commit, while we actually want "the remote branch tip of the commit we are checking out".

So I think the start point should be `origin/{*targetBranch}` if the repo is not a fork, else `fork/{*targetBranch}`.

You can use this script to see this behavior: https://gist.github.com/bduffany/da1ef7d857279e521a8dd14af199aeda

Reply:
Not relevant anymore now that hostedrunner always passes a target branch